src/wallet/bdk.rs

```diff
@@ -82,15 +82,21 @@ pub struct Wallet {
 /// Each kind represents an unhardened index value. Starting with External = 0.
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
 pub enum KeychainKind {
-    External = 0isize,
+    External,
     Internal,
+    Fidelity,
+    SwapCoin { id: u32 },
+    Contract,
 }
 
 impl KeychainKind {
     fn index_num(&self) -> u32 {
         match self {
             Self::External => 0,
             Self::Internal => 1,
+            Self::Fidelity => 2,
+            Self::SwapCoin { id } => 3 + id,
+            Self::Contract => 4,
```

Comment:
This won't work. When id=1, SwapCoin and Contract keychain index will be same. This will mess things up.